### PR TITLE
service_config::toml_merge handles toml::Tables (rebased w/ changed order)

### PR DESCRIPTION
**Pull request by robbkidd**
Rebased @metadave's `dp_toml_merge_table_fix` branch from #1185 and added on a fix for config value precedence in the merge. 

I'm not sure how to re-add the "types mismatch in config _values_" test, though.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/sup/src/service_config.rs [hunk 1]
```diff
@@ -345,20 +368,20 @@ impl Cfg {
     }
 
     fn to_toml(&self) -> toml::Value {
-        let mut left = toml::Table::new();
-        if let Some(toml::Value::Table(ref right)) = self.default {
-            left = toml_merge(&left, right);
+        let mut output_toml = toml::Table::new();
```
Comment (metadave) [inline components/sup/src/service_config.rs]: I think `output_toml` is more clear, nice!

Comment (metadave) [inline components/sup/src/service_config.rs]: actually, all the vars in this block have better names, thank you!

File: components/sup/src/service_config.rs [hunk 2]
```diff
@@ -698,6 +723,30 @@ mod test {
...
+        let default_config = "[server]
+            port = \"8080\"
+            shutdown-port = \"8005\"
+            redirect-port = \"8443\"";
+
+        let override_config = "server = { port=\"9090\" }";
+        let default_toml = toml::Parser::new(default_config).parse().unwrap();
+        let override_toml = toml::Parser::new(override_config).parse().unwrap();
+        {
+            let result = toml_merge(&override_toml, &default_toml);
```
Comment (metadave) [inline components/sup/src/service_config.rs]: I still think the order of the args to `toml_merge` is backwars, BUT, this PR fixes the issue and affects less code than switching the order of args. So, +1

**Comment by metadave** [comment]
Excellent!

I'll merge when Travis is green.

gif-keyboard-13605179485467846138

File: components/sup/src/service_config.rs (hunk 2 above)
Comment (robbkidd) [inline components/sup/src/service_config.rs]: Ya. I concur. This is a short term fix and there's probably goodness to be found in reworking the merge logic.

**Pull request merged**